### PR TITLE
Some clear guidelines for the legacy algs.

### PR DESCRIPTION
**Pull request by beldmit**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man7/migration_guide.pod [hunk 1]
```diff
@@ -101,6 +101,11 @@ should ensure that the legacy provider has been loaded. This can be achieved
 either programmatically or via configuration. See L<crypto(7)> man page for
 more information about providers.
 
+If the application code uses any of such algorithms via the EVP interface, the
+application should verify the result of the L<EVP_EncryptInit(3)>,
+L<EVP_EncryptInit_ex(3)>, and L<EVP_DigestInit(3)> functions. In case when
+the requested legacy algorithm is not available, these functions will fail.
```
Comment (t8m) [inline doc/man7/migration_guide.pod:107]: IMO this is giving false impression that non-legacy algorithms cannot fail. Applications should always check the return codes if they want to work correctly. So please adjust the text somewhat.

Comment (beldmit) [reply doc/man7/migration_guide.pod]: Fixed

**Review by t8m — APPROVED** [review]

File: doc/man7/migration_guide.pod [hunk 2]
```diff
@@ -60,6 +60,12 @@ There is however a dedicated C<install_fips> make target, which serves the
 special purpose of installing only the FIPS provider into an existing
 OpenSSL installation.
 
+Not all algorithms may be available for the application at a particular moment.
```
Comment (paulidale) [inline doc/man7/migration_guide.pod:63]: What about this alternative text:

> Due to the change to providers for implementations, it cannot be assumed that algorithms are always available.  For example, an application that uses a cipher or digest algorithm via the EVP interface will need to verify the result of the corresponding L<EVP_EncryptInit(3)>,
L<EVP_EncryptInit_ex(3)> or L<EVP_DigestInit(3)> function call.  These functions return a failure when the requested algorithm is not available.

Comment (t8m) [reply doc/man7/migration_guide.pod:63]: Well... the init() calls might have failed even on 1.1.1 in certain rare circumstances. So it was always a very good idea to check the return value.

Comment (t8m) [reply doc/man7/migration_guide.pod:63]: So I slightly prefer keeping the text as is.

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**